andela/mervels-bn-backend
# #168131417 Reset Password via email

#### What does this PR do?
Sends reset password link to users email.
User is able to reset their passwords using the link sent.
#### Description of Task to be completed?
- User requests for a password reset
- Check if User if has an account
- Send an email with a link to users email if account present
- Display error message if account missing
- User clicks on the link and thier prompted to enter new password
- If Link is not corrupted and correct password is provided then password is reset.
#### How should this be manually tested?
- Git clone this repo and checkout to this branch
- Run `npm install`
- Create a sendGrid account and acquire API key
- add senGrid API_KEY to the .env file as directed by .env.example
- add `localhost:3000` as baseUrl value in .env file
- Run `npm run start:dev`
- visit POST  http://localhost3000/api/v1/auth/requestPasswordReset via postman
- enter 
`{
	"email": "dkddabitswa@gmail.com"
}`
- check your email if a valid email is used
- Click on reset password in the email
- Copy the link from the browser and test it via postman
- for example PUT http://localhost3000/api/v1/auth/resetPassword/1/fdxgchvjxdgfhjdtfyug
- Enter new password atleast (1 number, 1 uppercase, 1lowercase)
`{
	"password": "d1wAsssss",
	"newPassword": "d1wAsssss"
}`
- Processed to login.
- You can also visit the documentation http://localhost:3000/api/docs to test
#### Any background context you want to provide?
- You need a sendGrid API key to send email
- The link returned in the data is to be used to reset new password.
- You can also use x-www-urlencoded data to test
- For tests, an email won't be sent since we are accessing the sendGrid sandbox 
- You need an internet connection to run tests
#### What are the relevant pivotal tracker stories?
[168131417](https://www.pivotaltracker.com/story/show/168131417)
#### Screenshots (if appropriate)
<img width="1430" alt="Screen Shot 2019-09-11 at 10 06 52" src="https://user-images.githubusercontent.com/26115429/64684831-18e6dd00-d486-11e9-9973-c2afd4bb2811.png">
<img width="1126" alt="Screen Shot 2019-09-11 at 10 08 05" src="https://user-images.githubusercontent.com/26115429/64684849-200deb00-d486-11e9-9722-ccbc7bad504b.png">
<img width="1128" alt="Screen Shot 2019-09-11 at 10 09 00" src="https://user-images.githubusercontent.com/26115429/64684858-2734f900-d486-11e9-91bd-a8e7508aae51.png">

#### Questions:
N/A
